knora/dsplib/utils/xml_upload.py

```diff
@@ -720,33 +757,62 @@ def _handle_upload_error(
 
     Args:
         err: error that was the cause of the abort
-        input_file: file name of the original XML file
         id2iri_mapping: mapping of ids from the XML file to IRIs in DSP (only successful uploads appear here)
         failed_uploads: resources that caused an error when uploading to DSP
         stashed_xml_texts: all xml texts that have been stashed
         stashed_resptr_props: all resptr props that have been stashed
+        proj_shortcode: shortcode of the project where the data is uploaded
+        onto_name: name of the ontology into which the data is uploaded
+        server: the server onto which the data is uploaded
+        save_location: path to the directory where dsp-tools can save logs (OS dependent)
 
     Returns:
         None
     """
 
     print(f'\n=========================================='
           f'\nxmlupload must be aborted because of an error')
-    timestamp_str = datetime.now().strftime("%Y%m%d-%H%M%S")
-
-    # write id2iri_mapping of the resources that are already in DSP
-    if len(id2iri_mapping) > 0:
-        _write_id2iri_mapping(input_file, id2iri_mapping, timestamp_str)
+    timestamp_str = datetime.now().strftime("%Y-%m-%d_%H%M%S")
+
+    server_substitutions = {
+        r"https?://": "",
+        r"^api\..+": "",
+        r":\d{4}/?$": "",
+        r"0.0.0.0": "localhost"
+    }
+    for pattern, repl in server_substitutions.items():
+        server = re.sub(pattern, repl, server)
+
+    if save_location == ".":
+        save_location_full = f"xmluploads/{server}/{proj_shortcode}/{onto_name}"
+    else:
+        save_location_full = f"{save_location}/xmluploads/{server}/{proj_shortcode}/{onto_name}"
+    os.makedirs(save_location_full, exist_ok=True)
```

Comment:
again, a Path object would make this much nicer (plus, if you can eliminate the `.` case, this will get simpler